base/applications/cleanmgr/dialog.c

```diff
@@ -0,0 +1,360 @@
+					COLOR_HIGHLIGHT : COLOR_WINDOW));
+
+			GetTextMetrics(lpdis->hDC, &tm);
+			y = (lpdis->rcItem.bottom + lpdis->rcItem.top - tm.tmHeight) / 2;
+			x = LOWORD(GetDialogBaseUnits()) / 4;
+
+			SendMessage(lpdis->hwndItem, CB_GETLBTEXT,
+				lpdis->itemID, (LPARAM)achTemp);
+
+
+			StringCchLength(achTemp, _countof(achTemp), &cch);
+
+			ExtTextOut(lpdis->hDC, CX_BITMAP + 2 * x, y,
+				ETO_CLIPPED | ETO_OPAQUE, &lpdis->rcItem,
+				achTemp, (UINT)cch, NULL);
+
+			SetTextColor(lpdis->hDC, clrForeground);
+			SetBkColor(lpdis->hDC, clrBackground);
+
+			HDC hdc = CreateCompatibleDC(lpdis->hDC);
+			if (hdc == NULL)
+				break;
+
+			SelectObject(hdc, hbmMask);
+			BitBlt(lpdis->hDC, x, lpdis->rcItem.top + 1,
+				CX_BITMAP, CY_BITMAP, hdc, 0, 0, SRCAND);
+
+			SelectObject(hdc, hbmIcon);
+			BitBlt(lpdis->hDC, x, lpdis->rcItem.top + 1,
+				CX_BITMAP, CY_BITMAP, hdc, 0, 0, SRCPAINT);
+
+			DeleteDC(hdc);
+
+			if (lpdis->itemState & ODS_FOCUS)
+				DrawFocusRect(lpdis->hDC, &lpdis->rcItem);
+
+			break;
+		}
+
+		case WM_COMMAND:
+			if(HIWORD(wParam) == CBN_SELCHANGE)
+			{
+				ItemIndex = SendMessageW(GetDlgItem(hwnd, IDC_DRIVE), CB_GETCURSEL, 0, 0);
+				if (ItemIndex == CB_ERR)
+				{
+					MessageBoxW(NULL, L"SendMessageW failed!", L"Error", MB_OK | MB_ICONERROR);
+					return FALSE;
+				}
+				SendMessageW(GetDlgItem(hwnd, IDC_DRIVE), CB_GETLBTEXT, (WPARAM)ItemIndex, (LPARAM)wcv.driveLetter);
+			}
+
+			switch(LOWORD(wParam))
+			{
+				case IDOK:
+					if (wcslen(wcv.driveLetter) == 0)
+					{
+						MessageBoxW(hwnd, L"Select a drive!", L"Warning", MB_OK | MB_ICONWARNING);
+						break;
+					}
+					DeleteObject(hbmDrive);
+					DeleteObject(hbmMask);
+					EndDialog(hwnd, IDOK);
+					break;
+				case IDCANCEL:
+					DeleteObject(hbmDrive);
+					DeleteObject(hbmMask);
+					EndDialog(hwnd, IDCANCEL);
+					break;
+            }
+			break;
+
+		case WM_CLOSE:
+			EndDialog(hwnd, IDCANCEL);
+			break;	
+
+		default:
+			return FALSE;
+    }
+	return TRUE;
+}
+
+BOOL CALLBACK ProgressDlgProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
+{
+	static HANDLE threadOBJ = NULL;
+	WCHAR tempText[MAX_PATH] = { 0 };
+	WCHAR fullText[MAX_PATH] = { 0 };
+
+	switch(Message)
+    {
+        case WM_INITDIALOG:
+			LoadStringW(GetModuleHandleW(NULL), IDS_SCAN, tempText, _countof(tempText));
+			StringCchPrintfW(fullText, _countof(fullText), tempText, wcv.driveLetter);
+			dv.hwndDlg = hwnd;
+			SetDlgItemTextW(hwnd, IDC_STATIC_SCAN, fullText);
+			threadOBJ = CreateThread(NULL, 0, &sizeCheck, NULL, 0, NULL);
+			return TRUE;
+		
+		case WM_NOTIFY:
+			return ThemeHandler(hwnd, (LPNMCUSTOMDRAW)lParam);
+
+		case WM_THEMECHANGED:
+			InvalidateRect(hwnd, NULL, FALSE);
+			break;
+
+		case WM_COMMAND:
+			switch (LOWORD(wParam))
+			{
+				case IDCANCEL:
+					CloseHandle(threadOBJ);
+					EndDialog(hwnd, IDCANCEL);
+					break;
+			}
+		case WM_CLOSE:
+			CloseHandle(threadOBJ);
+			EndDialog(hwnd, IDCANCEL);
```

Comment:
Don't call EndDialog here. WM_CLOSE is closing.

Reply:
@katahiromz Isn't WM_CLOSE message is called when the user prompts a program to close itself (by Alt + F4 or clicking Close button at the top) and we actually need to tell to destroy the dialog box (i.e. by using EndDialog function).